src/main.ts

```diff
@@ -1,16 +1,135 @@
-    core.debug(`Waiting ${ms} milliseconds ...`)
+    core.debug('got inside the action')
+
+    const pr = github.context.payload.pull_request
+    if (!pr) {
+      core.info('No pull request, exiting action...')
+      return
+    }
+    core.debug('got pull request')
+
+    const terraformPlan: TerraformPlan = JSON.parse(
+      core.getInput('terraform_plan_json')
+    )
+
+    const token = core.getInput('github_token')
+    const nwo = process.env['GITHUB_REPOSITORY'] || '/'
+    const [owner, repo] = nwo.split('/')
+    const octokit = github.getOctokit(token)
+
+    const toCreate = []
+    const toDelete = []
+    const toReplace = []
+    const toUpdate = []
+    for (const resourceChange of terraformPlan.resource_changes) {
+      core.debug(`resource: ${JSON.stringify(resourceChange)}`)
+      const actions = resourceChange.change.actions
+      const resourceName = `${resourceChange.type} - ${resourceChange.name}`
+      if (actions.length === 1 && actions.includes(Action.create)) {
+        toCreate.push(resourceName)
+      } else if (actions.length === 1 && actions.includes(Action.delete)) {
+        toDelete.push(resourceName)
+      } else if (
+        actions.length === 2 &&
+        actions.includes(Action.delete) &&
+        actions.includes(Action.create)
+      ) {
+        toReplace.push(resourceName)
+      } else if (actions.length === 1 && actions.includes(Action.update)) {
+        toUpdate.push(resourceName)
+      } else {
+        core.debug(`Not found? ${actions}`)
+      }
+    }
+
+    core.debug(`toCreate: ${toCreate}`)
+    core.debug(`toUpdate: ${toUpdate}`)
+    core.debug(`toReplace: ${toReplace}`)
+    core.debug(`toDelete: ${toDelete}`)
 
-    core.debug(new Date().toTimeString())
-    await wait(parseInt(ms, 10))
-    core.debug(new Date().toTimeString())
+    let body = `${commentPrefix}\n`
+    if (toCreate.length > 0) {
+      body += `will create ${toCreate.length} resources: \n`
+      for (const resource of toCreate) {
+        body += `- ${resource}`
+      }
+      body += '\n\n'
+    }
+    if (toUpdate.length > 0) {
+      body += `will update ${toUpdate.length} resources: \n`
+      for (const resource of toUpdate) {
+        body += `- ${resource}`
+      }
+      body += '\n\n'
+    }
+    if (toReplace.length > 0) {
+      body += `will replace (**delete** then create) ${toReplace.length} resources: \n`
+      for (const resource of toReplace) {
+        body += `- ${resource}`
+      }
+      body += '\n\n'
+    }
+    if (toDelete.length > 0) {
+      body += `will **delete** ${toDelete.length} resources: \n`
+      for (const resource of toDelete) {
+        body += `- ${resource}`
+      }
+      body += '\n\n'
+    }
+    if (
+      toCreate.length === 0 &&
+      toUpdate.length === 0 &&
+      toReplace.length === 0 &&
+      toDelete.length === 0
+    ) {
+      body += 'No changes'
+    } else {
+      // TODO find a way to link directly to job/step
```

Comment:
There's [an API for it](https://developer.github.com/v3/actions/workflow-jobs/#list-jobs-for-a-workflow-run). See the `html_url` in the response. If we're willing to standardize on a `name` for this action, we might even be able to link directly to the step.

Reply:
This is difficult, we can get a link to the workflow run, but to the specific job and the specific step, I don't know how we can do that simply. That API returns a list of jobs, if the workflow has multiple jobs, how do I know which one has the terraform plan? I guess I can require an input of the step_id of the terraform plan in this action so that it can maybe sort through the jobs' steps to find the step to link to?

Reply:
I'm fine if we punt on this for now. 🙂